lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2122,7 +2124,12 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   case OpInBoundsPtrAccessChain: {
     auto AC = static_cast<SPIRVAccessChainBase *>(BV);
     auto Base = transValue(AC->getBase(), F, BB);
-    Type *BaseTy = transType(AC->getBase()->getType()->getPointerElementType());
+    SPIRVType *BaseSPVTy = AC->getBase()->getType();
+    Type *BaseTy =
+        BaseSPVTy->isTypeVector()
+            ? transType(
+                  BaseSPVTy->getVectorComponentType()->getPointerElementType())
```

Comment:
Can this logic be safely moved inside getPointerElementType?

Reply:
It's a very good suggestion. But after some considering I'd say: no.

The reasoning, that it's safe to move it now, but I can imagine, that in the future some vendor, which is not aware or interested in our extension, can add another instructions, that are operating over pointers. The change you are suggesting would implicitly allow vector of pointers to be used by those instruction, which is incorrect.

And may be it's already the case.